eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -62,22 +86,24 @@ where
     /// Generate the private key
     pub fn generate_private_key(&self, rng: &mut ThreadRng) -> PrivateKey<C> {
         loop {
-            let f = self.sample_gaussian(rng);
-            let finv = f.inverse();
-
-            let Ok(finv) = finv else {
-                continue;
-            };
+            let f = self.sample_key(rng);
 
+            // TODO: document the equation that is being implemented here
             let mut priv_key = f.clone();
             priv_key *= C::t_as_coeff();
+
+            // Raw coefficient access must be followed by a truncation check.
             priv_key[0] += C::Coeff::one();
             priv_key.truncate_to_canonical_form();
 
             let priv_key_inv = priv_key.inverse();
 
-            if let Ok(_priv_key_inv) = priv_key_inv {
-                return PrivateKey { f, finv, priv_key };
+            if let Ok(priv_key_inv) = priv_key_inv {
+                return PrivateKey {
+                    f,
+                    priv_key_inv,
```

Comment:
This field was changed from `f_inv` to `priv_key_inv`, but the test is still checking `f_inv`.

This occasionally causes failures, particularly with tiny polynomials:
https://github.com/Inversed-Tech/eyelid/actions/runs/9072716142/job/24928644893?pr=82#step:5:31
So I’m not sure if we can merge failing tests to the main branch.

Can you update the test so it does what you expect?
I’m not sure what we need to do to test that the keys and encryption are correct, so I don’t want to merge without your input.

Reply:
I think we should not test an encryption scheme using tiny parameters. It is not secure when instantiated like that, an even correctness doesn't hold and weird things happen. We should worry making it pass the unit test because a corresponding situation doesn't happen for real parameters. Can we stop using tiny parameters for YASHE tests?

Reply:
This is now ticket #91.